ckb-importer/src/main/java/com/hartwig/serve/ckb/datamodel/treatmentapproaches/RelevantTreatmentApproaches.java

```diff
@@ -17,7 +18,10 @@ public abstract class RelevantTreatmentApproaches {
     public abstract int id();
 
     @Nullable
-    public abstract DrugClass drugClass();
+    public abstract DrugClassInfo drugClass();
```

Comment:
Having two nullable fields here means there's nothing in the code stopping someone setting both non-null.  Also, clients to this object can't make assumptions about whether its a drug based or therapy based, they need to always check.  And the alternative is possible, they could both be null. Does that make any sense? 

An alternative would be to have two classes DrugTreatmentApproach and TherapyTreatmentApproach, each with the appropriate "approach". They could extend a common superclass to reduce duplication. 

ActionableEvidenceFactory could then traverse both collections independently and without the nullable conditionals. 

I have not completely investigated the feasibility of doing this, maybe there is some context I'm missing. So please take as a suggestion for an improvement worth considering, not a mandatory change for approval.

Reply:
I have implement some logics with a interface `TreatmentApproachIntervation` which is an optional intervation. Some context: 
- DrugClassTreatmentApproach contains data and TherapyTreatmentApproach is nullable
- TherapyTreatmentApproach contains data and DrugClassTreatmentApproach is nullable
- both DrugClassTreatmentApproach and TherapyTreatmentApproach are nullabel

Reply:
Hey Lieke, not exactly what I had in mind for a couple of reasons:
- Empty marker interfaces should generally be avoided, they don't really have much use as you nearly always end up having to use instanceof/cast to figure out what they are.
- We still end up needing multiple checks on the nullability of the field to decide whether the approach is drug or therapy based.

I've created a new PR on top of this PR to illustrate: https://github.com/hartwigmedical/serve/pull/32. If you like it, you can merge that PR into this one.

Reply:
thanks for your suggestion 👍 You have made a @NotNull list instead of a nullable item. I will merge it to the main OSP-85 branch